ambry-api/src/test/java/com.github.ambry/rest/RestUtilsTest.java

```diff
@@ -549,12 +554,21 @@ public void getOperationOrBlobIdFromUriTest() throws JSONException, UnsupportedE
         String testPath = testCase.getKey();
         // skip the ones with no leading slash if prefix is not "". Otherwise they become -> "/prefixexpectedOp".
         if (prefixToTestOn.isEmpty() || testPath.startsWith("/")) {
-          String realTestPath = prefixToTestOn + testPath;
+          // if there is a signed path in uri, no prefix will be added into uri
+          String realTestPath = testPath.startsWith("/" + signedPath) ? testPath : prefixToTestOn + testPath;
           String expectedOutput = testCase.getValue();
           // skip the ones where the prefix will not be removed and the expected output is the blobId
           if (prefixesToRemove.contains(prefixToTestOn) || !expectedOutput.equals(blobId)) {
-            expectedOutput =
-                prefixesToRemove.contains(prefixToTestOn) ? expectedOutput : prefixToTestOn + expectedOutput;
+            if (!prefixesToRemove.contains(prefixToTestOn)) {
+              // remove query part if present
+              int endIndex = testPath.indexOf("?");
+              String processedTestPath = endIndex == -1 ? testPath : testPath.substring(0, endIndex);
+              // there are 2 cases to consider when prefix is not removed:
+              // 1. "/orNotToRemove/baseId" and expected output is "/baseId", because the first segment is considered as signed-path
+              // 2. "/orNotToRemove/" and expected output is "/orNotToRemove/", because there are only 2 segments (split by "/") and first segment won't be removed
```

Comment:
The URIs themself are incorrect, we need to reach an agreement on how to handle such URIs. 
There is one risky case we should pay attention to if using my currently implementation:

If the URI is `/orNotToRemove/baseId` for container without signed path validation, then it should throw `BadRequest` exception. However in current implementation, the `/orNotToRemove` is removed in `getOperationOrBlobIdFromUri`, we should add extra check in `postProcessRequest` to ensure this URI with invalid prefix is rejected.